src/com/untamedears/realisticbiomes/GrowthConfig.java

```diff
@@ -261,103 +238,65 @@ public boolean isPersistent() {
-		}
-		
-		// if the greenhouse effect does not ignore biome, fold the biome rate into the main rate directly
-		if (!greenhouseIgnoreBiome) {
-			rate *= environmentMultiplier;
-			environmentMultiplier = 1.0;
-		}
-		
-		// modulate the rate by the amount of sunlight recieved by this plant
-		int sunlightIntensity;
-		if (block.getType().isTransparent()) {
-			sunlightIntensity = block.getLightFromSky();
-		} else {
-			sunlightIntensity = block.getRelative(BlockFace.UP).getLightFromSky();
-		}
-		if (needsSunlight) {
-			environmentMultiplier *= Math.pow((sunlightIntensity / MAX_LIGHT_INTENSITY), 3.0);
-		}
-		// apply multiplier if the sunlight is not at maximum
-		if (sunlightIntensity < MAX_LIGHT_INTENSITY) {
-			environmentMultiplier *= notFullSunlightMultiplier;
-		}
-		
-		// if the crop block if fully lit, and the greenhouse rate would be an improvement
-		// over the current environment multiplier, then use the green house rate as the
-		// environment multiplier.
-		if(isGreenhouseEnabled && ( environmentMultiplier < greenhouseRate ) && ( block.getLightFromBlocks() == (MAX_LIGHT_INTENSITY - 1) ) ) {
-			// make sure it's a glowstone/lamp
+		if (isGreenhouseEnabled && block.getLightFromBlocks() == (MAX_LIGHT_INTENSITY - 1)) {
 			for( Vector vec : adjacentBlocks ) {
 				Material mat = block.getLocation().add(vec).getBlock().getType();
 				if( mat == Material.GLOWSTONE || mat == Material.REDSTONE_LAMP_ON ) {
-					environmentMultiplier = greenhouseRate;
+					rate *= greenhouseRate;
 					break;
 				}
 			}
 		}
-		
-		rate *= environmentMultiplier;
-		
-		// if the plant will be effected by irrigation, check if the block is at the correct level and
-		// check if nearby blocks are water blocks in river biomes
-		if (notIrrigatedMultiplier != 1.0) {
-			// determine if the block is near a water block in a river biome
-			boolean irrigated = false;
-			for( Vector vec : waterCheckBlocks ) {
-				Block waterBlock = block.getLocation().add(vec).getBlock();
-				Material mat = waterBlock.getType();
-				Biome biome = waterBlock.getBiome();
-				if((biome == Biome.RIVER || biome == Biome.FROZEN_RIVER) && (mat == Material.STATIONARY_WATER || mat == Material.WATER)) {
-					irrigated = true;
-					break;
-				}
+		else {
+			int sunlightIntensity;
+			if (block.getType().isTransparent()) {
+				sunlightIntensity = block.getLightFromSky();
+			} else {
+				sunlightIntensity = block.getRelative(BlockFace.UP).getLightFromSky();
 			}
-			
-			if (!irrigated)
-				rate *= notIrrigatedMultiplier;
-		}
-		
-		// check the depth of the required 'soil' and add a bonus
-		float soilBonus = 0.0f;
-		Block newBlock = block.getRelative(0,-soilLayerOffset,0);	
-		int soilCount = 0;
-		while (soilCount < soilMaxLayers) {
-			if (newBlock == null || !newBlock.getType().equals(soilMaterial)) {
-				break;
+			if (needsSunlight) {
+				rate *= Math.pow((sunlightIntensity / MAX_LIGHT_INTENSITY), 3.0);

```

Comment:
My instinct tell me to check for `if (needsSunlight && sunlightIntensity != MAX_LIGHT_INTENSITY)`, but someone that has more java-fu than me might know better.